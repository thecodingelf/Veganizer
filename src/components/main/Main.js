import React, { Component } from 'react'
import {
  View,
  Platform,
  Text
} from 'react-native'

const instructions = Platform.select({
    ios: 'More features coming soon!',
    android: 'More features coming soon!',
  })
  

  class Main extends Component {
    
    render() {

        const {
            username
        } = this.props.users

        return (
            <View /*style={containerStyle}*/>
                <Text /*style={welcomeStyle}*/>
                    Welcome to Kawa Demo! {username}
                </Text>
                <Text /*style={instructionsStyle}*/>
                    To get started, create your river.
                </Text>
                <Text /*style={instructionsStyle}*/>
                    {instructions}
                </Text>
            </View>
        );
    }
}

export default Main