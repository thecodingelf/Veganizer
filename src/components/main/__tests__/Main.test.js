import 'react-native';
import React from 'react';
import Main from '../Main';
import { Platform } from 'react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const instructions = Platform.select({
    ios: 'More features coming soon!',
    android: 'More features coming soon!',
  })


it('Renders correctly', () => {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Renders the view and text component', () => {
    const View = require('View');
    const Text = require('Text');
    
    const tree = renderer
        .create(<View><Text>{instructions}</Text></View>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});