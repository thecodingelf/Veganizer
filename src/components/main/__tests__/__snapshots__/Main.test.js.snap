// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<View>
  <Text>
    Welcome to Kawa Demo!
  </Text>
  <Text>
    To get started, create your river.
  </Text>
  <Text>
    More features coming soon!
  </Text>
</View>
`;

exports[`Renders the view and text component 1`] = `
<View>
  <Text>
    More features coming soon!
  </Text>
</View>
`;
